packages/pyright-scip/src/lib.ts

```diff
@@ -152,6 +218,26 @@ export function formatSnapshot(
             }
         }
 
+        // Check if any enclosing ranges start on this line
+        for (let rangeIndex = 0; rangeIndex < enclosingRanges.length; rangeIndex++) {
+            const enclosingRange = enclosingRanges[rangeIndex];
+
+            if (enclosingRange.range.start.line == lineNumber) {
+                // Switch the range to the open list
+                enclosingRanges.splice(rangeIndex, 1);
+                openEnclosingRanges.push(enclosingRange);
+
+                // Decrement the counter as an item was removed
+                rangeIndex -= 1;
+
+                pushEnclosingRange(enclosingRange);
+
+                continue;
+            }
+
+            break;
+        }
```

Comment:
I think the logic here and below would be simplified quite a bit if instead of the sort earlier + lookup here, you instead did something like creating two lists outside this loop:

```
const enclosingRangeStarts = Array.from(Array(inputs.lines.length), () => [])
const enclosingRangeEnds = Array.from(Array(inputs.lines.length), () => [])
for (const enclosingRange of enclosingRanges) {
  // Push entry to start, and ends array based on line numbers
}
```

Later, when iterating over the lines, you can sort the sub-arrays at index `i` based on the line, and consume that directly, instead of having to do a lookup through the list of ranges.

There wouldn't be any need for element removal or +1/-1 logic.